src/validator/CreditCardValidator.java

```diff
@@ -0,0 +1,62 @@
+package validator;
+
+import java.util.ArrayList;
+
+public class CreditCardValidator implements CreditCardChecking {
+
+	@Override
+	public boolean checkValidity(CreditCardVendor ccv, String number) {
+		ArrayList<Integer> chars = ccv.getNumberOfChars();
+		ArrayList<String> masks = ccv.getIIN();
+		
+		boolean result1 = checkLengthCorrectness(number, chars);
+		boolean result2 = checkIINMaskCorrectness (number, masks);
+		boolean result3 = checkLuhnalgorithmCorrectness(number);
+		
+		return result1 && result2 && result3;
+	}
+	
+	private boolean checkLengthCorrectness(String number, ArrayList<Integer> chars){
+		boolean result = false;
+		for (int i = 0; i<chars.size();i++){
+			if (chars.get(i) == number.length())
+				result = true;
+		}
+		return result;
```

Comment:
If in line 23 we will terminate the loop, here result will be always false. So we can simplify to return false here. And now we can remove 
`boolean result = false; `
from line 20

Reply:
In summary in this method we check if chars List contain number.length(). So we can replace for loop by:
`return chars.contains(number.length());`